CONTRIBUTING.md

```diff
@@ -1,52 +1,122 @@
+communicating so, could result into maintainers not looking into your patch,
+unless they CI that CI pass.\
+_Desired behavior:_ Write a message saying roughly the following "The CI failure
+is unrelated", so maintainers will fix it for you.
 
-Once your PR is open, you can ask for a review by a maintainer of your platform. Winit's merging policy
-is that a PR must be approved by at least two maintainers of winit before being merged, including
-at least a maintainer of the platform (a maintainer making a PR themselves counts as approving it).
+_Event:_ Collaborator requested review on your PR.\
+_Desired behavior:_ Once you address the request, you should re-request review
+with the github's UI. If you don't agree with what maintainer suggested, you
+should object that and re-request the review. That will indicate that the
+ball is on maintainer's side.
 
-Once your PR is deemed ready, the merging maintainer will take care of resolving conflicts in the
-`changelog` module (but you must resolve other conflicts yourself). Doing this requires that you check the
-"give contributors write access to the branch" checkbox when creating the PR.
+_Event:_ You've opened a PR, but maintainer shortly after commented that they
+want to work on that themselves.\
+_Desired behavior:_ Discuss with the maintainer regarding their plans if they
+were not outlined in the initial response, because such response means that they
+are not interested in reviewing your code. Such thing could happen when
+underestimating complexity of the task you're solving or when your patch
+mandate certain downstream designs. In general the maintainer should close your
+PR in order to prevent work being done on it.
 
-## Maintainers
+[prs]: https://github.com/rust-windowing/winit/pulls
+[issues]: https://github.com/rust-windowing/winit/issues
+[matrix]: https://matrix.to/#/#rust-windowing:matrix.org
 
-The current maintainers for each platform are listed in the [CODEOWNERS](.github/CODEOWNERS) file.
+## Maintainers
 
-## Release process
+Winit has plenty of maintainers with different backgrounds, different time
+available to work on Winit, and reasons to be winit maintainer in the first
+place. To ensure that winit's code quality is not decreasing over time and to
+make it easier to teach new maintainers the _winit way of doing things_ the
+common policies and routines are defined in this section.
+
+The current maintainers for each platform are listed in [this file][CODEOWNERS].
+
+### Contributions handling
+
+The maintainers must ensure that the external contributions meet the winit's
+quality standards. If it's not, it **is the maintainer's responsibility** to
+bring it on par, which includes:
+
+  - Ensure that formatting is consistent and `CHANGELOG` messages are clear
+    for the end users.
+  - Improve the commit message, so it'll be easier for other maintainers to
+    understand the motivation without going through all the discussions on the
+    particular patch/PR.
```

Comment:
I still know I'm going to forget to do this, but I'll try

Reply:
Don't worry about it, I can fix it if needed